src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,225 @@
+
+const fetchTokenFallbackGasEstimates = () =>
+  // https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/util/TokenDepositUtils.ts#L45-L51
+  // Use hardcoded gas estimate values
+  // Values set by looking at a couple of different ERC-20 deposits
+  bn(240_000)
+
+export async function getTradeQuote(
+  input: GetEvmTradeQuoteInput,
+): Promise<Result<TradeQuote, SwapErrorRight>> {
+  const {
+    chainId,
+    sellAsset,
+    buyAsset,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    supportsEIP1559,
+    receiveAddress,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+  } = input
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
+    throw new Error(`Arbitrum Bridge only supports EVM chains`)
+  }
+
+  const adapter = assertGetEvmChainAdapter(chainId)
+
+  const l2Network = await getL2Network(arbitrum.id)
+
+  const isDeposit = sellAsset.chainId === ethChainId
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+  const isTokenBridge = !isEthBridge
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+  const erc20L1Address = fromAssetId((isDeposit ? sellAsset : buyAsset).assetId).assetReference
+  const erc20L2Address = fromAssetId((isDeposit ? buyAsset : sellAsset).assetId).assetReference
+  const l1Provider = getEthersV5Provider(KnownChainIds.EthereumMainnet)
+  const l2Provider = getEthersV5Provider(KnownChainIds.ArbitrumMainnet)
+
+  if (isTokenBridge) {
+    // Since our related assets list isn't exhaustive and won't cut it to determine the L1 <-> L2 mapping, we double check that the bridge is valid
+    // by checking against Arbitrum bridge's own mappings, which uses different sources (Coingecko, Gemini, Uni and its own lists at the time of writing)
+    const arbitrumBridgeErc20L2Address = await (bridger as Erc20Bridger).getL2ERC20Address(
+      erc20L1Address,
+      l1Provider,
+    )
+    const arbitrumBridgeErc20L1Address = await (bridger as Erc20Bridger).getL1ERC20Address(
+      erc20L2Address,
+      l2Provider,
+    )
+
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L1Address), getAddress(erc20L1Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L1 ERC20 address: ${erc20L1Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+    if (!isAddressEqual(getAddress(arbitrumBridgeErc20L2Address), getAddress(erc20L2Address))) {
+      return Err(
+        makeSwapErrorRight({
+          message: `[ArbitrumBridge: tradeQuote] - Invalid L2 ERC20 address: ${erc20L2Address}`,
+          code: TradeQuoteError.UnsupportedTradePair,
+        }),
+      )
+    }
+  }
+
+  if (sellAsset.assetId === usdcAssetId || sellAsset.assetId === usdcOnArbitrumAssetId) {
```

Comment:
just verifying the intention here is we don't even want to support usdc -> usdc.e with a notification? also, first glance at the docs is vague, but does usdc.e -> usdc (eth) require CCTP as well? Or is that only eth -> arb?

Reply:
USDC <-> USDC.e *should* work, seemingly that's what would be automagically used when using Arbitrum Bridge (i.e the matching L1 for USDC.e is USDC, and the matching L2 for USDC is USDC.e) - disabled all for the sake of simplicity.
Not sure if both directions are supported though, will give this some more testing and comment back here, we should probably be able to at least bring back USDC.e bridging, though that's obviously assuming users select USDC.e as a swapping asset and not USDC, which is definitely valid for users that already have some USDC.e, but probably not what users wanting to bridge USDC from mainnet would expect to receive, and hence select.

Reply:
Removed the USDC condition in 75ccaa2ca1 - we indeed don't support cctp bridges, but that will be automagically detected with our current checks.

- Mainnet native USDC -> Arb native USDC is invalid

<img width="1034" alt="image" src="https://github.com/shapeshift/web/assets/17035424/2f103bd0-42bc-4a42-9c26-e100a046384b">

- Arb native USDC -> Mainnet native USDC is invalid

<img width="1043" alt="image" src="https://github.com/shapeshift/web/assets/17035424/5538312a-3c19-45ca-beb7-6dedb387b950">

- USDC.e -> USDC

<img width="1054" alt="image" src="https://github.com/shapeshift/web/assets/17035424/d6f3edb8-d4e4-4463-9b39-9b137e07c937">
<img width="574" alt="image" src="https://github.com/shapeshift/web/assets/17035424/3c04d58f-5a8a-410e-8c67-2289897a34b4">
<img width="524" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9a142839-4971-4904-9c3a-1313422d6208">

- USDC -> USDC.e

Untested because of insufficient funds, but here it is at quote time:

<img width="1046" alt="image" src="https://github.com/shapeshift/web/assets/17035424/7ede3e1f-5c34-48d4-b4c0-facaa7eaa271">